examples/web/main.go

```diff
@@ -47,6 +48,20 @@ func saveReportfunc(ctx agent.CallbackContext, llmResponse *model.LLMResponse, l
 	return llmResponse, llmResponseError
 }
 
+type AuthInterceptor struct {
+}
+
+// Intercept implements a2asrv.RequestContextInterceptor.
+func (a *AuthInterceptor) Intercept(ctx context.Context, reqCtx *a2asrv.RequestContext) (context.Context, error) {
+	if reqCtx.Metadata == nil {
+		reqCtx.Metadata = make(map[string]any)
+	}
+	reqCtx.Metadata["adk_user_id"] = "user"
+	return ctx, nil
+}
+
+var _ a2asrv.RequestContextInterceptor = (*AuthInterceptor)(nil)
```

Comment:
not needed. `a2asrv.WithRequestContextInterceptor` already requires the interface as a parameter type.